examples/search-knn.js

```diff
@@ -0,0 +1,85 @@
+// This example demonstrates how to use RediSearch to index and query data 
+// stored in Redis hashes using vector similarity search.
+
+import { createClient, SchemaFieldTypes } from 'redis';
+
+const client = createClient();
+
+await client.connect();
+
+// Create an index...
+try {
+  // Documentation: https://redis.io/docs/stack/search/reference/vectors/
+  await client.sendCommand([
+    'FT.CREATE',
+    'idx:vectors',
+    'SCHEMA',
+    'vector',
+    'VECTOR',
+    'HNSW',
+    '6',
+    'TYPE',
+    'FLOAT32',
+    'DIM',
+    '512',
+    'DISTANCE_METRIC',
+    'COSINE',
+  ]);
+} catch (e) {
+  if (e.message === 'Index already exists') {
+    console.log('Index exists already, skipped creation.');
+  } else {
+    // Something went wrong, perhaps RediSearch isn't installed...
+    console.error(e);
+    process.exit(1);
+  }
+}
+
+// Add some sample data...
+// https://redis.io/commands/hset/
+await Promise.all([
+  client.hSet('noderedis:vectors:1', { vector: [0.4, -0.325, 4.3] },
```

Comment:
Do we need to encode these vectors with `Buffer`? How should that be done exactly?